serve/methods/gas.go

```diff
@@ -0,0 +1,112 @@
+package methods
+
+import (
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/std"
+)
+
+type gasFeeTotalInfo struct {
+	Low         int64
+	High        int64
+	TotalAmount int64
+	TotalCount  int64
+}
+
+// GetGasPricesByBlock calculates the gas price statistics (low, high, average)
+// for a single block.
+func GetGasPricesByBlock(block *types.Block) ([]*GasPrice, error) {
+	blocks := []*types.Block{block}
+
+	return GetGasPricesByBlocks(blocks)
+}
+
+// GetGasPricesByBlocks calculates the gas price statistics (low, high, average)
+// for multiple blocks.
+func GetGasPricesByBlocks(blocks []*types.Block) ([]*GasPrice, error) {
+	gasFeeInfoMap := make(map[string]*gasFeeTotalInfo)
+
+	for _, block := range blocks {
+		blockGasFeeInfo := calculateGasFeePerBlock(block)
+
+		for denom, gasFeeInfo := range blockGasFeeInfo {
+			currentGasFeeInfo := gasFeeInfoMap[denom]
+			gasFeeInfoMap[denom] = calculateGasFee(currentGasFeeInfo, gasFeeInfo)
+		}
+	}
+
+	return calculateGasPrices(gasFeeInfoMap), nil
+}
+
+// calculateGasFeePerBlock processes all transactions in a single block to compute
+// gas fee statistics (low, high, total amount, total count) for each gas fee denomination.
+func calculateGasFeePerBlock(block *types.Block) map[string]*gasFeeTotalInfo {
+	gasFeeInfo := make(map[string]*gasFeeTotalInfo)
+
+	for _, t := range block.Txs {
+		var stdTx std.Tx
+		if err := amino.Unmarshal(t, &stdTx); err != nil {
+			continue
+		}
+
+		denom := stdTx.Fee.GasFee.Denom
+		amount := stdTx.Fee.GasFee.Amount
+
+		info := gasFeeInfo[denom]
+		if info == nil {
+			info = &gasFeeTotalInfo{}
+			gasFeeInfo[denom] = info
+		}
+
+		info.Low = minInt64WithDefault(info.Low, amount)
+		info.High = max(info.High, amount)
+		info.TotalAmount += amount
+		info.TotalCount++
+	}
+
+	return gasFeeInfo
+}
+
+// calculateGasFee merges the gas fee statistics from a block into the global statistics.
+func calculateGasFee(currentInfo, blockInfo *gasFeeTotalInfo) *gasFeeTotalInfo {
```

Comment:
Nitpick: you modify the actual values in the func (you pass in pointers), and re-return a single pointer value, which is not needed

`func calculateGasFee(currentInfo, blockInfo gasFeeTotalInfo)`, or rename this func to be `modifyActiveInfo`

Reply:
Renamed the function`modifyAggregatedInfo`.
https://github.com/gnolang/tx-indexer/pull/120/commits/d302eacac3035d97fa338e50d87a13432c4a04aa